### DISCUSSION TITLE
How do I get started with ebpf-go?

### DISCUSSION THREAD

**Discussion by kolinfluence** [cilium/ebpf#882] [Q&A] [answered]
how do u guys learn to use ebf cilium? any tutorials on this?

**Comment by saucer-man** [comment]
question too

**Reply by Appstek-ebpf-linux-webassembly** [reply]
Thanks

**Comment by ti-mo** [comment, marked as answer]
A few pointers:
- Michael Kashin has an excellent introduction at https://networkop.co.uk/post/2021-03-ebpf-intro
- [The README's Getting Started section](https://github.com/cilium/ebpf#getting-started) points to the [`examples/`](https://github.com/cilium/ebpf/tree/master/examples) folder, which has many jump-off points for building custom tools using the library
- There's no shortage of importers (https://pkg.go.dev/github.com/cilium/ebpf?tab=importedby) to take inspiration from
- In 2023, we're planning on publishing a site with documentation covering various topics on eBPF programming, built around the library
- For any ebpf-related questions, [StackOverflow](https://stackoverflow.com/questions/tagged/ebpf) and `#ebpf`/`#ebpf-go` on Cilium's Slack.

**Reply by wanghonglei5181** [reply]
> * 2023

is the web site ready? when ?

**Reply by ti-mo** [reply]
> is the web site ready?

It is not.

> when ?

As stated, somewhere this year.

**Reply by duanhaowei** [reply]
> > is the web site ready?
> 
> It is not.
> 
> > when ?
> 
> As stated, somewhere this year.

ebpf-go is great !! hope website come soon
